### PR TITLE
Allow disabling key connector if no user is enrolled

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Allows disabling of key connector as long as no organization user is enrolled.

Resolves https://app.asana.com/0/1201363553288890/1201363553288929

## Testing requirements

Verify key connector can be disabled if no user is enrolled, and verify it cannot if a user is enrolled.

## Before you submit
- [ ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [x] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: test/Core.Test/Services/SsoConfigServiceTests.cs [hunk 1]
```diff
@@ -96,16 +97,53 @@ public async Task SaveAsync_PreventDisablingKeyConnector(SutProvider<SsoConfigSe
...
 
-            Assert.Contains("KeyConnector cannot be disabled at this moment.", exception.Message);
+            Assert.Contains("Key Connector cannot be disabled at this moment.", exception.Message);
 
             await sutProvider.GetDependency<ISsoConfigRepository>().DidNotReceiveWithAnyArgs()
                 .UpsertAsync(default);
         }
 
+        [Theory, CustomAutoData(typeof(SutProviderCustomization))]
+        public async Task SaveAsync_AllowDisablingKeyConnectorNoUserUsing(SutProvider<SsoConfigService> sutProvider,
```
Comment (djsmith85) [inline test/Core.Test/Services/SsoConfigServiceTests.cs:113]:
```suggestion
        public async Task SaveAsync_AllowDisablingKeyConnectorWhenNoUserIsUsingIt(SutProvider<SsoConfigService> sutProvider,
```

File: src/Core/Services/Implementations/SsoConfigService.cs [hunk 2]
```diff
@@ -42,15 +46,23 @@ public async Task SaveAsync(SsoConfig config)
...
             {
-                throw new BadRequestException("KeyConnector cannot be disabled at this moment.");
+                throw new BadRequestException("Key Connector cannot be disabled at this moment.");
             }
 
             await LogEventsAsync(config, oldConfig);
             await _ssoConfigRepository.UpsertAsync(config);
         }
 
+        private async Task<bool> UserHasKeyConnectorEnabledAsync(Guid organizationId)
```
Comment (djsmith85) [inline src/Core/Services/Implementations/SsoConfigService.cs:59]:
```suggestion
        private async Task<bool> HasAOrgUserEnabledKeyConnectorAsync(Guid organizationId)
```

**Review by djsmith85 — APPROVED** [review]
Just some naming suggestions

File: src/Core/Services/Implementations/SsoConfigService.cs (hunk 2 above)
Comment (Hinton) [reply src/Core/Services/Implementations/SsoConfigService.cs:59]: @djsmith85 What do you think of `AnyOrgUserHasKeyConnectorEnabledAsync` I think including `Any` makes it even more explicit.

Comment (djsmith85) [reply src/Core/Services/Implementations/SsoConfigService.cs]: @Hinton That works for me. I tend to prefix Has/Is/Does with bools, hence the suggestion

Comment (Hinton) [reply src/Core/Services/Implementations/SsoConfigService.cs]: I think as long as it's formulated as a question with a yes/no response it's fine.

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**